operator/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -628,10 +654,84 @@ public boolean equals(Object other) {
       verifyNoAlternateSecretNamespaceSpecified();
       addMissingModelConfigMap(kubernetesResources);
       verifyIstioExposingDefaultChannel();
+      verifyIntrospectorJobName();
 
       return failures;
     }
 
+    private void verifyIntrospectorJobName() {
+      // K8S adds a 5 character suffix to an introspector job name
+      if (LegalNames.toJobIntrospectorName(getDomainUid()).length()
+          > LegalNames.LEGAL_DNS_LABEL_NAME_MAX_LENGTH - 5) {
+        failures.add(DomainValidationMessages.exceedMaxIntrospectorJobName(
+            getDomainUid(),
+            LegalNames.toJobIntrospectorName(getDomainUid()),
+            LegalNames.LEGAL_DNS_LABEL_NAME_MAX_LENGTH - 5));
+      }
+    }
+
+    private void verifyGeneratedResourceNames(WlsDomainConfig wlsDomainConfig) {
+      checkGeneratedServerServiceName(wlsDomainConfig.getAdminServerName(), -1);
+      if (isExternalServiceConfigured(getSpec())) {
+        checkGeneratedExternalServiceName(wlsDomainConfig.getAdminServerName());
+      }
+
+      // domain level serverConfigs do not contain servers in dynamic clusters
+      wlsDomainConfig.getServerConfigs()
+          .values()
+          .stream()
+          .map(WlsServerConfig::getName)
+          .forEach(serverName -> checkGeneratedServerServiceName(serverName, -1));
+      wlsDomainConfig.getClusterConfigs()
+          .values()
+          .iterator()
+          .forEachRemaining(wlsClusterConfig
+              // serverConfigs contains configured and dynamic servers in the cluster
+              -> wlsClusterConfig.getServerConfigs().forEach(wlsServerConfig
+                  -> this.checkGeneratedServerServiceName(
+                      wlsServerConfig.getName(), wlsClusterConfig.getServerConfigs().size())));
+      wlsDomainConfig.getClusterConfigs()
+          .values()
+          .iterator()
+          .forEachRemaining(wlsClusterConfig -> this.checkGeneratedClusterServiceName(wlsClusterConfig.getName()));
+    }
+
+    private void checkGeneratedExternalServiceName(String adminServerName) {
+      if (LegalNames.toExternalServiceName(getDomainUid(), adminServerName).length()
+          > LegalNames.LEGAL_DNS_LABEL_NAME_MAX_LENGTH) {
+        failures.add(DomainValidationMessages.exceedMaxExternalServiceName(
+            getDomainUid(),
+            adminServerName,
+            LegalNames.toExternalServiceName(getDomainUid(), adminServerName),
+            LegalNames.LEGAL_DNS_LABEL_NAME_MAX_LENGTH));
+      }
+    }
+
+    private void checkGeneratedServerServiceName(String serverName, int clusterSize) {
+      int limit = LegalNames.LEGAL_DNS_LABEL_NAME_MAX_LENGTH;
+      if (Main.Namespaces.isClusterSizePaddingValidation() && clusterSize > 0) {
+        limit = clusterSize >= 10 ? limit - 1 : limit - 2;
```

Comment:
The limit should not be adjusted downward for clusters greater than size 99.

Reply:
Make sense. thanks.